src/ethereum_test_tools/vm/opcode.py

```diff
@@ -25,6 +26,15 @@ def _get_int_size(n: int) -> int:
 _push_opcodes_byte_list = [bytes([0x5F + x]) for x in range(33)]
 
 
+class SpecialCodes(Enum):
```

Comment:
I think that what we're doing here is more akin to a macro maybe?
```suggestion
class Macro(Enum):
```

Reply:
Another thing is that we could leave the `class Opcodes` without any changes and define this further below as it's own separate class:

```python
class Macro(bytes):
    """
    Represents... etc..
    """
    def __call__(self, *) -> bytes:
        return self
```

```python
class Macros(Macro, Enum):
    """
    Enum containing all macros
    """
    OOG = Opcodes.SHA3(0, 100000000000)
```

So I think we should not need the `special_code` parameter in the `Opcode` class.

The only reason why we would need the `class Macro` is to make it compatible with `Opcodes` by making it callable.

The downside of this is typing: If any test explicitly wants to create, e.g. a list of opcodes that includes a macro, they need to define it as `List[Opcodes | Macros]`, but it might not be a big deal.

Reply:
Yes, I also was thinking to nest the class

Reply:
I managed to encode it in the same Op enum